### ISSUE TITLE
Payment due 2024-09-23[$250] [P2P Distance] Add RBR logic for the bottom up flow

### ISSUE THREAD

**Issue by neil-marcellini** [Expensify/App#46753]
Implement the plan in [this design doc section](https://docs.google.com/document/d/15GQhipcqOJ69MbFA54KRUcWxjICURHYcfTlAb3eaVEQ/edit#heading=h.uucfqvh11n53) to add RBR logic for the bottom up flow.

Please branch off of this PR [[App] feat: Edit Distance Rate flow](https://github.com/Expensify/App/pull/40021) if it hasn't been merged yet.
<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01c65559d6ec0b66ea</li>
        <li>Upwork Job ID: 1819470334389076893</li>
        <li>Last Price Increase: 2024-08-02</li>
<li>Automatic offers: </li>
<ul>
<li>paultsimura | Contributor | 103375763</li>
<li>brunovjk | Contributor | 103375768</li>
</ul></li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @zanyrenney</details>

**Comment by neil-marcellini** [comment]
Ah shoot. Luckily it looks like we are allowed to share the test account credentials. Try setting up a bank account using the information provided in the [steps here](https://github.com/Expensify/App/issues/45278#issue-2403195928). Lmk if that works or if you have further issues.

Oh, I see that maybe you have already done that. What are the conditions for App to show the bank account as a payment option?

**Comment by brunovjk** [comment]
Not overdue -- @paultsimura let me know if you need any help :D

**Comment by brunovjk** [comment]
@paultsimura, I know these challenges can be tough, but you’ve faced similar ones before and come through. You’ve got this!

**Comment by neil-marcellini** [comment]
Great vibes here, I love the positivity 😄

> @neil-marcellini is the "Rate not valid for this workspace" (taken from the doc) a verified copy?

I can't remember honestly. Let's go ahead assuming it is, but I'll also request a copy check to make sure everyone is happy with it.

May I please have a copy check for the violation error message we'll show in NewDot for the bottom up flow? You can see an example in this [design doc section](https://docs.google.com/document/d/15GQhipcqOJ69MbFA54KRUcWxjICURHYcfTlAb3eaVEQ/edit#heading=h.8o0j25va12ai). Is the copy we have in the mockup good?

**Comment by zanyrenney** [comment]
hey @ryanschaffer, we're looking for a verification on this copy suggestion, please?

> "Rate not valid for this workspace" (taken from the doc) 

from @neil-marcellini [above](https://github.com/Expensify/App/issues/46753#issuecomment-2276545283)
> May I please have a copy check for the violation error message we'll show in NewDot for the bottom up flow? You can see an example in this [design doc section](https://docs.google.com/document/d/15GQhipcqOJ69MbFA54KRUcWxjICURHYcfTlAb3eaVEQ/edit#heading=h.8o0j25va12ai). Is the copy we have in the mockup good?

Thanks!

**Comment by jamesdeanexpensify** [comment]
Sounds good to me, thanks!

**Comment by jamesdeanexpensify** [comment]
Wait, actually we should probably say something more affirmative like "Please choose a valid workspace rate"

**Comment by jamesdeanexpensify** [comment]
Got it, thanks! To keep things moving then, I think I'm OK with the original suggestion: `"Rate not valid for this workspace"`

**Comment by neil-marcellini** [comment]
I will start working on some backend fixes now

**Comment by neil-marcellini** [comment]
Head's up, I'll be working 50% for the rest of this week and next 🌲

**Comment by zanyrenney** [comment]
Hey @neil-marcellini any update on this one? Appreciate you're working 50% but if we can keep this updated according to the KSV2 cadence, that would be great. 

@paultsimura if you can also contribute to the updates whilst Neil is working part time, that would really help. Cheers!

**Comment by zanyrenney** [comment]
awesome work @paultsimura thank you!

**Comment by zanyrenney** [comment]
@neil-marcellini looks like we might be waiting on you for the BE changes before @paultsimura can finish off his portion.

**Comment by neil-marcellini** [comment]
> Hey @neil-marcellini any update on this one? Appreciate you're working 50% but if we can keep this updated according to the KSV2 cadence, that would be great.
> 
> @paultsimura if you can also contribute to the updates whilst Neil is working part time, that would really help. Cheers!

Good idea. I will drop this to a weekly since P2P distance is quite low priority as I understand it.


> @neil-marcellini looks like we might be waiting on you for the BE changes before @paultsimura can finish off his portion.

Yes that is the case. I have been working on higher priority items. I will set a reminder to work on this on Thursday, if not sooner.

**Comment by neil-marcellini** [comment]
I got started by writing a failing test for this flow. To be clear, I'm focusing on the edge case where a p2p distance expense is created towards the end of the year, then the default USD rate for workspaces is updated for the new year, after which the receiver pays the expense on a business bank account, creating a workspace. Then the submitter notices the violation and updates to the workspace default rate. The goal of the PR is to make sure the P2P distance rate data is properly updated/cleared and the violation is also cleared. I might need a Web-E PR to clear the violation. There might be other flows that could cause a similar violation, but this one is the most familiar to me so I'll focus on it.

**Comment by neil-marcellini** [comment]
I worked on this a little more today. Fixed some mistakes in the test and implemented the solution. Now the test is erroring and I'll have to debug it.

**Comment by neil-marcellini** [comment]
Haven't had time recently, working on higher priority items

**Comment by neil-marcellini** [comment]
I finally got the backend PR up for review! I'm going to go ahead and start testing and potentially adding on to @paultsimura's [PR](https://github.com/Expensify/App/pull/47136) because the flow we want to test requires changing the backend code part way through the test steps.

**Comment by neil-marcellini** [comment]
@zanyrenney when it comes to payment time I think we should give @paultsimura the full amount, even though I jumped in and wrote some of the PR myself. The part he worked on was definitely substantial. I chose to jump in because the flow we want to test requires changing the backend as part of one of the steps, so it was the most efficient way to finish it off.

**Comment by zanyrenney** [comment]
payment due next week, not overdue.

**Comment by brunovjk** [comment]
I didn't review the PR - https://github.com/Expensify/App/pull/47136#issuecomment-2344721228

**Comment by zanyrenney** [comment]
**payment summary**

$250 to @paultsimura paid via upwork
no payment for review because this was reviewed internally by @youssef-lr and @neil-marcellini
